cylc/flow/task_job_mgr.py

```diff
@@ -227,17 +233,30 @@ def submit_task_jobs(self, suite, itasks, curve_auth,
         for install_target, itasks in sorted(auth_itasks.items()):
             # Re-fetch a copy of platform
             platform = itasks[0].platform
-            is_init = self.task_remote_mgr.remote_init(
-                platform, curve_auth, client_pub_key_dir)
-            if is_init is None:
-                # Remote is waiting to be initialised
+            if install_target not in self.task_remote_mgr.remote_init_map.keys(
+            ) or self.task_remote_mgr.remote_init_map[install_target] is None:
+                self.task_remote_mgr.remote_init(
+                    platform, curve_auth, client_pub_key_dir)
```

Comment:
I *think* this branch could do with splitting in two to avoid calling `remote_init` when not necessary though might be missing something.

Reply:
So, now remote_init_map install targets should never be mapped to none. I have removed this condition.